packages/clerk-js/src/core/clerk.ts

```diff
@@ -1388,11 +1388,23 @@ export class Clerk implements ClerkInterface {
   public __internal_navigateToTaskIfAvailable = async ({
     redirectUrlComplete,
   }: __internal_NavigateToTaskIfAvailableParams = {}): Promise<void> => {
+    const onBeforeSetActive: SetActiveHook =
+      typeof window !== 'undefined' && typeof window.__unstable__onBeforeSetActive === 'function'
+        ? window.__unstable__onBeforeSetActive
+        : noop;
+
+    const onAfterSetActive: SetActiveHook =
+      typeof window !== 'undefined' && typeof window.__unstable__onAfterSetActive === 'function'
+        ? window.__unstable__onAfterSetActive
+        : noop;
+
     const session = this.session;
     if (!session || !this.environment) {
       return;
     }
 
+    await onBeforeSetActive();
```

Comment:
This fixes an issue where, upon redirection to `redirectUrlComplete`, the middleware was still cached with the previous JWT claim where `sessionStatus` was `pending`, then an infinite loop of requests was occurring - the middleware kept redirecting back to the task.


https://github.com/user-attachments/assets/96460970-8155-4993-8014-776e6bc54b74